timelock-impl/src/main/java/com/palantir/atlasdb/timelock/lock/NamedMinTimestampTrackerImpl.java

```diff
@@ -58,14 +77,17 @@ public synchronized void unlock(long timestamp, UUID requestId) {
                     SafeArg.of("requestId", requestId),
                     SafeArg.of("currentHolder", holdersByTimestamp.get(timestamp)));
         }
+        numLocksHeld.decrementAndGet();
     }
 
     @Override
     public synchronized Optional<Long> getMinimumTimestamp() {
         if (holdersByTimestamp.isEmpty()) {
             return Optional.empty();
         }
-        return Optional.of(holdersByTimestamp.firstKey());
+        Long minimum = holdersByTimestamp.firstKey();
```

Comment:
And in general, I'm less bought in to the idea of this metric right now / care about it the least for the initial milestone, so at least split this out into a separate PR, so we can ship the counter first.

Reply:
I will spin out a PR **only** for the shortfall thing

Reply:
https://github.com/palantir/atlasdb/pull/7402